exercises/ansible_rhel/1.6-templates/README.md

```diff
@@ -3,161 +3,126 @@
+Exercise 1.5 introduces Jinja2 templating within Ansible, a powerful feature for generating dynamic files from templates. You'll learn how to craft templates that incorporate host-specific data, enabling the creation of tailored configuration files for each managed host.
 
 ## Guide
 
-### Step 1 - Using Templates in Playbooks
+### Step 1 - Introduction to Jinja2 Templating
+
+Ansible leverages Jinja2, a widely-used templating language for Python, allowing dynamic content generation within files. This capability is particularly useful for configuring files that must differ from host to host.
+
+### Step 2 - Crafting Your First Template
 
-When a template for a file has been created, it can be deployed to the managed hosts using the `template` module, which supports the transfer of a local file from the control node to the managed hosts.
+Templates end with a `.j2` extension and mix static content with dynamic placeholders enclosed in `{{ }}`.
 
-As an example of using templates you will change the motd file to contain host-specific data.
+In the following example, let's create a template for the Message of the Day (MOTD) that includes dynamic host information.
 
-First create the directory `templates` to hold template resources in `~/ansible-files/`:
+#### Set Up the Template Directory:
+
+Ensure a templates directory exists within your lab_inventory directory to organize your templates.
 
 ```bash
-[student@ansible-1 ansible-files]$ mkdir templates
+mkdir -p ~/lab_inventory/templates
 ```
 
-Then in the `~/ansible-files/templates/` directory create the template file `motd-facts.j2`:
+#### Develop the MOTD Template:
 
-<!-- {% raw %} -->
+Create a file named `motd.j2` in the templates directory with the following content:
 
-```html+jinja
+```jinja
 Welcome to {{ ansible_hostname }}.
-{{ ansible_distribution }} {{ ansible_distribution_version}}
-deployed on {{ ansible_architecture }} architecture.
+OS: {{ ansible_distribution }} {{ ansible_distribution_version }}
+Architecture: {{ ansible_architecture }}
 ```
 
-<!-- {% endraw %} -->
+This template dynamically displays the hostname, OS distribution, version, and architecture of each managed host.
+
+### Step 3 - Deploying the Template with a Playbook
 
-The template file contains the basic text that will later be copied over. It also contains variables which will be replaced on the target machines individually.
+Utilize the `ansible.builtin.template` module in a playbook to distribute and render the template across your managed hosts.
 
-Next we need a playbook to use this template. In the `~/ansible-files/` directory create the Playbook `motd-facts.yml`:
+Modify the `system_setup.yml` Playbook with the following content:
 
 ```yaml
 ---
-- name: Fill motd file with host data
-  hosts: node1
+- name: Basic System Setup
+  hosts: all
   become: true
   tasks:
-    - ansible.builtin.template:
-        src: motd-facts.j2
+.
```

Comment:
3 extra dots (".") in 3 lines below the src: motd-facts.j2

Reply:
The extra dots are there to show that its just a snippet as showing the entire file would make the example longer than it needs to be.